### ISSUE TITLE
[False-Positive] One-time use variables + references

### ISSUE THREAD

**Issue by fruitl00p** [kalessil/phpinspectionsea#1285]
Subject | Details
-- | --
Issue type | False Positive
Plugin | Php Inspections (EA Ultimate), 2.0.14
Language level | n/a

## Current behaviour:
First of all: Not absolutely sure if it's a false positive, but for me it is.

Please see the following example code:
```php
class A
{
    private $tree = [
        'flat' => [],
        'structured' => [],
    ];

    private function buildTree()
    {
        $nodes = json_decode(json_encode($this->repository->find()), true);
        foreach ($nodes as & $node) {
            // Add the node to our associative array using it's id as key,
            // but check to see if the $node as already found via a earlier child... :)
            if (isset($this->tree['flat'][$node['id']])) {
                $node['children'] = &$this->tree['flat'][$node['id']]['children'];
            }
            $this->tree['flat'][$node['id']] =& $node;

            // Add empty placeholder for children
            if (!isset($node['children'])) {
                $node['children'] = [];
            }
            // It it's a root node, we add it directly to the tree
            if (empty($node['parentId'])) {
                $this->tree['structured'][$node['id']] =& $node;
            } else {
                // It was not a root node, add this node as a reference in the parent.
                $this->tree['flat'][$node['parentId']]['children'][$node['id']] =& $node;
            }
            unset($node['intro'], $node['subTitle'], $node['content'], $node['parentId'], $node['sliderImageIds']);
        }
        unset($node);
    }
}
```
From the example it might not be obvious, but this creates a nested tree from an array of 'rows' (i.e. from a database) in both a structured and flat list. (all with references to the flat list) Thus merging `$nodes` into the foreach wouldn't work as the `$node` is a reference that's added / referenced during the building of the tree... (i know, as i write this sounds complicated)

In any case: this seems like a false positive to me based on the `&node` in the foreach statement...

## Suggested behaviour
Do not report in this case, as the reference to an item in the array prohibits the inline-ing of the original array...

**Comment by kalessil** [comment]
Fixed!
